### PR TITLE
Simplify Rakefile

### PR DESCRIPTION
**Pull request by hsbt**
# Description:

I removed some duplicated tasks in the current Rakefile.
______________

# Tasks:

- [ ] Describe the problem / feature
- [ ] Write tests
- [ ] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -46,11 +46,6 @@ RDoc::Task.new :rdoc => 'docs', :clobber_rdoc => 'clobber_docs' do |doc|
   doc.rdoc_dir = 'doc'
 end
 
-desc "Install gems needed to run the tests"
-task :install_test_deps => :clean do
-  sh "gem install minitest -v '~> 5.0'"
```
Comment (deivid-rodriguez) [inline Rakefile]: The other place where this is defined is 

https://github.com/rubygems/rubygems/blob/a0b3d351043e824819b3d66ce3e7d6958b4ba252/util/ci#L55

Right?

Maybe we can install move that line here and call the rake task from there?

File: Rakefile [hunk 2]
```diff
@@ -352,11 +347,6 @@ end
 
 # Misc Tasks ---------------------------------------------------------
 
-desc "Cleanup trailing whitespace"
-task :whitespace do
-  system 'find . -not \( -name .svn -prune -o -name .git -prune \) -type f -print0 | xargs -0 sed -i "" -E "s/[[:space:]]*$//"'
-end
-
```
Comment (deivid-rodriguez) [inline Rakefile]: Why is this task duplicated? You mean by rubocop? Technically this task works on all project files, so if we add some traling whitespace to the README, this task will remove it.

Comment (hsbt) [reply Rakefile]: Fmm, Your comment is right. But I never invoke this task in the past releases.

File: Rakefile (hunk 1 above)
Comment (hsbt) [reply Rakefile]: I update it for using `rake setup` in `util/ci`

File: Rakefile (hunk 2 above)
Comment (deivid-rodriguez) [reply Rakefile]: Right. We should either remove it or use it, yeah. Maybe add it to CI? I don't have a strong opinion, whatever you think it's best works for me :+1:.

**Comment by hsbt** [comment]
@bundlerbot r+

**Pull request merged**